### ISSUE TITLE
Any suggestions on how to train a good D3Feat model?

### ISSUE THREAD

**Issue by liuQuan98** [XuyangBai/D3Feat#40]
Hi Xuyang
Hello, thank you for your excellent work! I am currently trying to train D3Feat on KITTI, both with your original dataset and my modified KITTI dataset (with different frame pairs). But with the default setting, I cannot get satisfying results with my modified dataset (test registration accuracy around 42%, way worse than your pretrained model on KITTI).

I must confess that this is confusing, and is probably caused by my unintentional code tweaks rather than false training strategies. I have been running reference experiments on freshly downloaded master version D3Feat as well to rule out that possibillity, but that will take days and I am in a hurry. I wonder if there are any suggestions or code changes to train a good model like the pretrained one on KITTI?

best
Quan

**Comment by XuyangBai** [comment]
Hi, thanks for your interest in our work. I cannot tell what is the reason for the low regsitration performance given limited information. Belows are some general suggessions:
1. If you are really in a hurry, you may finetune the released weight on KITTI on your modified dataset. The network should converge very fast in such cases.
2. You can monitor the training process through tensorboard to see whether overfitting or other unpredictable things happends. The `accuracy` printed out is not equal to the feature matching recall in paper, this value represent the percentage of `positive pair distance < negative pair distance` , which I use to moniter the training process. I wonder whether you interpret it incorrectly.


Best,
Xuyang

**Comment by liuQuan98** [comment]
Hi Xuyang
Thanks a lot for your timely reply! I will try finetuning the pretrained weight. 

Best
Quan
